### PR TITLE
Np 1500 bugfix get doi requests empty list

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java [hunk 1]
```diff
@@ -123,7 +122,8 @@ public void handleRequestReturnsNotFoundWhenPublicationDoesNotExist() throws IOE
         assertThat(response.getStatusCode(), is(equalTo(HttpStatus.SC_NOT_FOUND)));
 
         assertThat(details.getDetail(),
-            containsString(String.format(PUBLICATION_ID_NOT_FOUND_ERROR_FORMAT, notExistingPublicationIdentifier)));
+            containsString(DynamoDBDoiRequestsService.PUBLICATION_NOT_FOUND_ERROR_MESSAGE
```
Comment (norrs) [inline update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java:125]: Hm, we want to use qualified import right?

File: update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java [hunk 2]
```diff
@@ -254,16 +255,16 @@ private DoiRequestSummary readPublicationDirectlyFromDynamo(UUID id) throws NotF
...
             .withStatus(DoiRequestStatus.APPROVED)
             .build();
-        return new DoiRequestSummary(
-            publication.getIdentifier(),
-            publication.getOwner(),
-            includedDoiRequest,
-            publication.getEntityDescription());
+        publication.setDoiRequest(includedDoiRequest);
+        // copy actual modified date because it is set in Publication class by uncontrollable clock.
+        publication.setModifiedDate(actualDoiRequestSummary.getPublicationModifiedDate());
```
Comment (norrs) [inline update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java:266]: You can also assert that the data is within 10 seconds around the test. 
But it makes it hard to do a full assert equals between objects, so we want to be able to inject a fake clock down the line.

File: find-doi-requests/src/main/java/no/unit/nva/doi/requests/FindDoiRequestsHandler.java [hunk 3]
```diff
@@ -52,14 +51,22 @@ public FindDoiRequestsHandler() {
...
     public FindDoiRequestsHandler(DoiRequestsService doiRequestsService, Environment environment) {
         super(Void.class, environment, logger);
         this.doiRequestsService = doiRequestsService;
     }
 
+    @Override
+    protected DoiRequestsResponse processInput(Void input, String apiGatewayInputString, Context context)
+        throws ApiGatewayException {
+        logger.info("Whole event below");
+        logger.info(apiGatewayInputString);
```
Comment (norrs) [inline find-doi-requests/src/main/java/no/unit/nva/doi/requests/FindDoiRequestsHandler.java:66]: if we want this type of logging, maybe just delegate the responsibility to the ApiGateway class that can be turned on and off by a configuration value?

Comment (norrs) [reply find-doi-requests/src/main/java/no/unit/nva/doi/requests/FindDoiRequestsHandler.java:66]: And maybe not info as a logger level. trace most likley.

Comment (axthosarouris) [reply find-doi-requests/src/main/java/no/unit/nva/doi/requests/FindDoiRequestsHandler.java:66]: This actually should be deleted :)

File: update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java (hunk 2 above)
Comment (axthosarouris) [reply update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java:266]: I think we have to revisit how to add functionality closer to the data model

File: create-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/CreateDoiRequestHandlerTest.java [hunk 4]
```diff
@@ -120,7 +116,7 @@ public void handleRequestSavesDoiRequestToPublicationWhenPublicationIdIsNotEmpty
         CreateDoiRequest doiRequest = createDoiRequest(publication);
 
         InputStream input = createRequest(doiRequest, validUsername(publication));
-        ByteArrayOutputStream output = outpuStream();
+        ByteArrayOutputStream output = outputStream();
```
Comment (axthosarouris) [inline create-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/CreateDoiRequestHandlerTest.java:119]: fixed typo

File: create-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/CreateDoiRequestHandlerTest.java [hunk 5]
```diff
@@ -162,6 +158,10 @@ public void handleRequestReturnsConflictErrorWhenDoiRequestAlreadyExists()
         assertThat(problem.getDetail(), containsString(DynamoDBDoiRequestsService.DOI_ALREADY_EXISTS_ERROR));
     }
 
+    private String validUsername(Publication publication) {
```
Comment (axthosarouris) [inline create-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/CreateDoiRequestHandlerTest.java:161]: moved from line 111

File: create-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/CreateDoiRequestHandlerTest.java [hunk 6]
```diff
@@ -188,11 +188,8 @@ private DoiRequestSummary expectedDoiRequestSummary(Publication publication) {
             .withDate(mockNow)
             .withStatus(DoiRequestStatus.REQUESTED)
             .build();
-        return new DoiRequestSummary(
-            publication.getIdentifier(),
-            publication.getOwner(),
-            includedDoiRequest,
-            publication.getEntityDescription());
+        publication.setDoiRequest(includedDoiRequest);
+        return DoiRequestSummary.fromPublication(publication);
```
Comment (axthosarouris) [inline create-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/CreateDoiRequestHandlerTest.java:192]: DoiRequestSummary will be removed in subsequent PR

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/api/model/responses/DoiRequestIdentity.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
+package no.unit.nva.doi.requests.api.model.responses;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Objects;
+import java.util.UUID;
+import nva.commons.utils.JacocoGenerated;
+
+public class DoiRequestIdentity {
```
Comment (axthosarouris) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/api/model/responses/DoiRequestIdentity.java:9]: DoiRequestIdentity will be removed in subsequent PR

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/api/model/responses/DoiRequestSummary.java [hunk 8]
```diff
@@ -1,83 +1,77 @@
...
+@JsonTypeInfo(use = Id.NAME, property = "type")
+@SuppressWarnings("PMD.ExcessivePublicCount")
 public class DoiRequestSummary {
 
-    private final DoiRequestStatus doiRequestStatus;
-    private final Instant doiRequestDate;
-    private final UUID publicationIdentifier;
-    private final String publicationTitle;
-    private final String publicationOwner;
+    public static final String LOG_SERIALIZATION_ERROR_MESSAGE = "Could not serialize object:";
```
Comment (axthosarouris) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/api/model/responses/DoiRequestSummary.java:27]: DoiRequestSummary will be removed in subsequent PR

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/model/AbstractDoiRequest.java [hunk 9]
```diff
@@ -10,7 +10,6 @@
 public abstract class AbstractDoiRequest {
 
     public static final String INVALID_PUBLICATION_ID_ERROR = "Invalid publication id: ";
-    public static final String PUBLICATION_ID_NOT_FOUND_ERROR_FORMAT = "Publication with identifier %s not found.";
```
Comment (axthosarouris) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/model/AbstractDoiRequest.java:13]: removed duplicated String

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java [hunk 10]
```diff
@@ -97,23 +96,18 @@ public DynamoDBDoiRequestsService(AmazonDynamoDB client,
...
-        RangeKeyCondition rangeKeyCondition = new RangeKeyCondition(DOI_REQUEST_STATUS_DATE);
-        rangeKeyCondition.beginsWith(status.toString());
-        ItemCollection<QueryOutcome> outcome;
-        try {
-            outcome = doiRequestsIndex.query(PUBLISHER_ID, publisher.toString(), rangeKeyCondition);
-        } catch (Exception e) {
-            throw new DynamoDBException(ERROR_READING_FROM_TABLE, e);
-        }
-        return parseJsonToDoiRequestSummaries(outcome);
+        List<Publication> publications = fetchPublicationsByPublisherAndStatus(publisher, status);
```
Comment (axthosarouris) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:99]: refactored the method for better understanding

File: doi-requests-commons/src/test/java/no/unit/nva/doi/requests/api/model/responses/DoiRequestSummaryTest.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+import java.time.Instant;
+import java.util.UUID;
+import no.unit.nva.model.DoiRequest;
+import nva.commons.utils.JsonUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+
+class DoiRequestSummaryTest {
```
Comment (axthosarouris) [inline doi-requests-commons/src/test/java/no/unit/nva/doi/requests/api/model/responses/DoiRequestSummaryTest.java:12]: will be removed in subsequent PR

File: test-module/src/main/java/no/unit/nva/doi/requests/util/DoiRequestsDynamoDBLocal.java [hunk 12]
```diff
@@ -123,15 +122,7 @@ protected Publication getPublication(String tableName, UUID publicationId, Insta
...
-        return new Projection()
-            .withProjectionType(ProjectionType.INCLUDE)
-            .withNonKeyAttributes(TABLE_HASH_KEY,
-                CREATED_DATE,
-                TABLE_SORT_KEY,
-                ENTITY_DESCRIPTION,
-                DOI_REQUEST_FIELD_NAME,
-                STATUS,
-                OWNER);
+        return new Projection().withProjectionType(ProjectionType.ALL);
```
Comment (axthosarouris) [inline test-module/src/main/java/no/unit/nva/doi/requests/util/DoiRequestsDynamoDBLocal.java:125]: DoiRequest index returns all publication fields

File: find-doi-requests/src/main/java/no/unit/nva/doi/requests/FindDoiRequestsHandler.java (hunk 3 above)
Comment (axthosarouris) [reply find-doi-requests/src/main/java/no/unit/nva/doi/requests/FindDoiRequestsHandler.java]: deleted.

File: update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java (hunk 2 above)
Comment (axthosarouris) [reply update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java]: @norrs is there any specific action you would like here?

**Review by brinxmat — APPROVED** [review]

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/api/model/responses/DoiRequestSummary.java [hunk 13]
```diff
@@ -1,83 +1,77 @@
...
-        return publicationIdentifier;
-    }
-
-    public String getPublicationTitle() {
-        return publicationTitle;
-    }
-
-    public String getPublicationOwner() {
-        return publicationOwner;
+    private static String extractTitle(Publication publication) {
```
Comment (norrs) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/api/model/responses/DoiRequestSummary.java:69]: Why don't we move this as a getter on publication? Very odd this ask object about information etc.  I feel we should be able to ask Publication directly about its title. Not having to extract it and do stuff.

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/api/model/responses/DoiRequestSummary.java [hunk 14]
```diff
@@ -86,17 +80,56 @@ public boolean equals(Object o) {
...
+        return doiRequest;
+    }
+
+    public String getPublisherId() {
+        return publisherId;
+    }
+
+    @Override
+    public String toString() {
+        return attempt(this::toJsonString).orElse(this::failedSerializationWarning);
```
Comment (norrs) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/api/model/responses/DoiRequestSummary.java:122]: Clever.

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java [hunk 15]
```diff
@@ -97,23 +96,18 @@ public DynamoDBDoiRequestsService(AmazonDynamoDB client,
...
     }
 
     @Override
-    public List<DoiRequestSummary> findDoiRequestsByStatusAndOwner(URI publisher, DoiRequestStatus status, String owner)
+    public List<DoiRequestSummary> findDoiRequestsByStatusAndOwner(URI publisher, DoiRequestStatus status, String user)
         throws ApiGatewayException {
         return findDoiRequestsByStatus(publisher, status)
-            .stream()
-            .filter(doiRequestSummary -> doiRequestSummary.getPublicationOwner().equals(owner))
+            .stream().parallel()
```
Comment (norrs) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:109]: newline `parallel()`

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java [hunk 16]
```diff
@@ -97,23 +96,18 @@ public DynamoDBDoiRequestsService(AmazonDynamoDB client,
...
-        return parseJsonToDoiRequestSummaries(outcome);
+        List<Publication> publications = fetchPublicationsByPublisherAndStatus(publisher, status);
+        return publications
+            .stream().parallel()
+            .map(DoiRequestSummary::fromPublication).collect(Collectors.toList());
     }
 
     @Override
-    public List<DoiRequestSummary> findDoiRequestsByStatusAndOwner(URI publisher, DoiRequestStatus status, String owner)
+    public List<DoiRequestSummary> findDoiRequestsByStatusAndOwner(URI publisher, DoiRequestStatus status, String user)
```
Comment (norrs) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:106]: You can do later but ideally again a business method that belongs to `Publication` really.

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java [hunk 17]
```diff
@@ -145,34 +138,60 @@ public void updateDoiRequest(UUID publicationID, DoiRequestStatus requestedStatu
         putItem(publication);
     }
 
+    private boolean belongsToUser(String user, String publicationOwner) {
+        return nonNull(publicationOwner) && publicationOwner.equals(user);
+    }
+
+    private List<Publication> fetchPublicationsByPublisherAndStatus(URI publisher, DoiRequestStatus status)
```
Comment (norrs) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:145]: All this code has the weirdness of not simply telling the object what to do, you extract information and you have to combine and tweak around with data. 

Maybe doesnt belong in `DynamoDBDoiRequestService` which seems to be a storage service interface for accessing doi status stored in dynamodb.  

`Publication` has 2 mixed roles with this, one for being business object and one for being a DTO for the Dynamo backend? 

Responsibility concern comes  into mind.

**Review by norrs — COMMENTED** [review]
If this is a quick fix for getting the bug solved, we can rather travel back and clean code afterwards, just define a task for it so we are able to pick it up later in a sprint. @axthosarouris

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/api/model/responses/DoiRequestSummary.java (hunk 13 above)
Comment (axthosarouris) [reply doi-requests-commons/src/main/java/no/unit/nva/doi/requests/api/model/responses/DoiRequestSummary.java:69]: This class will cease to exist in the following PRs.

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java (hunk 16 above)
Comment (axthosarouris) [reply doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:106]: Not really. The misunderstanding comes from the fact that a DoiRequestSummary is basically a publication. So a Publication cannot return a list of Publications

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java (hunk 17 above)
Comment (axthosarouris) [reply doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:145]: One of the things that will happen in the next PR is that we will remove some classes and we will use Publication for DTO for now (in the scope of the bugfix). We do need to define the way to construct DTOs DAOs and BOs but this is outside the scope of this bugifx.

Comment (axthosarouris) [reply doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:145]: Created a bug for class responsibility problem: https://unit.atlassian.net/browse/NP-1573

**Review by norrs — APPROVED** [review]
For getting the bug fix out, go a head and ship. Shipping is more important.

**Pull request merged**